src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,38 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAXIMUM_SIZE = 10;
+    private final K[] key;
+    private final V[] value;
+    private int size;
+
+    public @SuppressWarnings("unchecked") StorageImpl() {
+        key = (K[]) new Object[MAXIMUM_SIZE];
+        value = (V[]) new Object[MAXIMUM_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            this.value[size - 1] = value;
+        } else {
+            this.key[size] = key;
+            this.value[size++] = value;
+        }
```

Comment:
let's fix this for better readability
```suggestion
            this.value[size] = value;
            size++;
        }
```